### PR TITLE
escape in json

### PR DESCRIPTION
**Pull request by ohadbitt**

### CODE REVIEW

File: connector/src/main/scala/com/microsoft/kusto/spark/datasink/CountingCsvWriter.scala [hunk 1]
```diff
@@ -33,13 +33,40 @@ case class CountingCsvWriter(out: Writer) {
...
+            bytesCounter += 1
+          } else if (c == '\b'){
+            out.write("\\b")
+            bytesCounter += 1
+          } else if ( c == '\f') {
+            out.write("\\f")
+            bytesCounter += 1
+          } else if (c == '\t'){
+            out.write("\\t")
+            bytesCounter += 1
```
Comment (danield137) [inline connector/src/main/scala/com/microsoft/kusto/spark/datasink/CountingCsvWriter.scala]: you have \t twice.
maybe have a collection of escapable chars and iterate them?
```java 
if (nested) {
    for (c <- str) {
        for ( escapable <- ['\t', '\n'...] {
            if c == esacapeable {
                     out.write('\' + c);
             }
        }
    }
}
```

File: connector/src/main/scala/com/microsoft/kusto/spark/datasink/CountingCsvWriter.scala [hunk 2]
```diff
@@ -33,13 +33,40 @@ case class CountingCsvWriter(out: Writer) {
 
       bytesCounter += 2
 
-      for (c <- str) {
-        if (c == '"') {
-          out.write("\"\"")
-          bytesCounter += 1
+      // Json does not allow special characters
+      if(nested){
```
Comment (danield137) [inline connector/src/main/scala/com/microsoft/kusto/spark/datasink/CountingCsvWriter.scala]: > Json does not allow special chars

I'm not sure I follow the logic. this function is called writeStringField, and is in a class called CsvWriter , 
why would you get a json?

Maybe we should handle this bad state else where?

Comment (ohadbitt) [reply connector/src/main/scala/com/microsoft/kusto/spark/datasink/CountingCsvWriter.scala]: Kusto dynamic is json

File: connector/src/main/scala/com/microsoft/kusto/spark/datasink/CountingCsvWriter.scala (hunk 1 above)
Comment (ohadbitt) [reply connector/src/main/scala/com/microsoft/kusto/spark/datasink/CountingCsvWriter.scala]: This wont work as theres a differnce between
'\r' and '\\r'
One is a special char , the later is slash followed by the letter r

Comment (ohadbitt) [reply connector/src/main/scala/com/microsoft/kusto/spark/datasink/CountingCsvWriter.scala]: This wont work as theres a differnce between
'\r' and '\\r'
One is a special char , the later is slash followed by the letter r

File: connector/src/main/scala/com/microsoft/kusto/spark/datasink/CountingCsvWriter.scala (hunk 2 above)
Comment (vladikbr) [reply connector/src/main/scala/com/microsoft/kusto/spark/datasink/CountingCsvWriter.scala]: This is the writeStringField method in class CountingCsvWriter class.
This has absolutely nothing to do with Json.

If it needs to enclose the field in quotes, it should.
If it needs to double the escaping sequences, it should.

You can tale a look at how the C# CsvWriter works and handles JSON string fields for reference.

Comment (ohadbitt) [reply connector/src/main/scala/com/microsoft/kusto/spark/datasink/CountingCsvWriter.scala]: I did 
This more resambles our JsonEscapedCsvWriter besides my addition of \f and \b
Do you suggest that the user will have to deal with it?

**Pull request merged**